jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1293,6 +1295,13 @@ public Method getDeclaredMethod(String name, Class<?>... parameterTypes) throws
 	 */
 	Class<?> declaringClass = cachedDeclaringClass == ClassReflectNullPlaceHolder.class ? null : cachedDeclaringClass;
 	if (declaringClass == null) {
+		if (cachedCheckInnerClassAttr == false) {
```

Comment:
Why not use `!` here instead of the explicit comparison?

Reply:
Agreed and fixed as suggested.